compiler/rustc_borrowck/src/region_infer/opaque_types.rs

```diff
@@ -66,85 +66,60 @@ impl<'tcx> RegionInferenceContext<'tcx> {
     ) -> FxIndexMap<LocalDefId, OpaqueHiddenType<'tcx>> {
```

Comment:
GH doesn't allow comments on unmodified code, but I have a question about the `infer_opaque_types` function's documentation above: will this doc comment still be accurate after the changes in this PR? 

(At least it seems the "which has no `external_name` in which case we use `'empty` as the" sentence was missed when `ReEmpty` was removed last year)

I don't know this resolving of opaque type regions at all, but I have the feeling that, if that high-level description doesn't need changes when the mapping to external names has been rewritten, then it's a probable sign that the body of the function itself could use more comments explaining what it does and why (e.g. the reasons why this required changing how the SCC representatives were picked in the first place)?

Reply:
I've missed this doc comment. It indeed needs a rewrite, in addition to documenting the new changes.

Reply:
This is still unaddressed

Reply:
Oh, ReEmpty is still there. give me a sec.